packages/strapi-admin/ee/admin/components/Roles/Permissions/ContentTypes/ContentTypesRow/ComponentsAttributes/ComponentAttributeRow.js

```diff
@@ -195,13 +193,15 @@ const ComponentAttributeRow = ({ attribute, index, numberOfAttributes, recursive
             </CollapseLabel>
           </RowStyle>
           <PermissionWrapper disabled={isSuperAdmin || (attribute.required && !isCollapsable)}>
-            {ATTRIBUTES_PERMISSIONS_ACTIONS.map(action => (
+            {staticAttributeActions.map(action => (
               <PermissionCheckbox
-                disabled={isSuperAdmin || (attribute.required && !isCollapsable)}
+                disabled={
+                  isSuperAdmin || (isCreateAndRequired(attribute, action) && !isCollapsable)
+                }
                 key={`${attribute.attributeName}-${action}`}
-                onChange={() => handleCheck(`${contentManagerPermissionPrefix}.${action}`)}
-                someChecked={someChecked(`${contentManagerPermissionPrefix}.${action}`)}
-                value={isChecked(`${contentManagerPermissionPrefix}.${action}`)}
+                onChange={() => handleCheck(action)}
```

Comment:
`onChange={e => handleCheck(e, action)}` // it might be worth to create a component here.